scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,862 @@
+            throws IOException {
+        try {
+            boolean alreadyExistingChannel = deleteChannel(originator, taskid, name);
+            Channel newChannel = new Channel();
+            if (isPersistent) {
+                persistedChannels.put(name, newChannel);
+                logWithContext(taskid, null, "Created new persistent channel " + QUOTE + name + QUOTE, Level.INFO);
+                recordManager.commit();
+            } else {
+                logWithContext(taskid, null, "Created new memory channel " + QUOTE + name + QUOTE, Level.INFO);
+                inMemoryChannels.put(name, newChannel);
+            }
+            return !alreadyExistingChannel;
+        } catch (InvalidChannelException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public boolean deleteChannel(String originator, TaskId taskid, String name) throws IOException {
+        try {
+            if (persistedChannels.containsKey(name)) {
+                persistedChannels.remove(name);
+                logWithContext(taskid, null, "Deleted persistent channel " + QUOTE + name + QUOTE, Level.INFO);
+                recordManager.commit();
+                return true;
+            } else if (inMemoryChannels.containsKey(name)) {
+                inMemoryChannels.remove(name);
+                logWithContext(taskid, null, "Deleted memory channel " + QUOTE + name + QUOTE, Level.INFO);
+                return true;
+            } else {
+                return false;
+            }
+        } catch (InvalidChannelException e) {
+            throw new IllegalStateException(e);
+        }
+    }
+
+    @Override
+    public boolean channelExists(String originator, TaskId taskid, String name) {
+        return inMemoryChannels.containsKey(name) || persistedChannels.containsKey(name);
+    }
+
+    @Override
+    public boolean createChannelIfAbsent(String originator, TaskId taskid, String name, boolean isPersistent)
+            throws IOException {
+        if (inMemoryChannels.containsKey(name) || persistedChannels.containsKey(name)) {
+            return false;
+        } else {
+            createChannel(originator, taskid, name, isPersistent);
+            return true;
+        }
+    }
+
+    /**
+     * Commits pending operations on the given channel
+     * @param channel channel name
+     * @throws IOException
+     */
+    private void commitIfNeeded(String channel) throws IOException {
+        if (persistedChannels.containsKey(channel)) {
+            // Record Manager mark as dirty (uncommited) entries which have be modified via a put call
+            // Thus, such operation as persistedChannels.get(channel).dosomething() will not be committed
+            // by the following trick, we mark the entry as dirty and commit
+            persistedChannels.put(channel, persistedChannels.get(channel));
+            recordManager.commit();
```

Comment:
Instead of calling `commit` each time we work on the persisted channels (at the risk of forgetting to call it), we could have a handler that takes the action to perform and that commits automatically at the end of the action.